### PR TITLE
Replace 'discovered_master' with 'discovered_cluster_manager' in 'GET Cat Health' API

### PR DESCRIPTION
**Pull request by tlfeng**
### Description
- Replace 'discovered_master' with 'discovered_cluster_manager' in the table header of [GET Cat Health]( (https://opensearch.org/docs/latest/opensearch/rest-api/cat/cat-health/)) API, to promote inclusive language.
- Add `dcm` as the alias for the table header `discovered_cluster_manager`
- Add `discovered_master` as the alias for the table header `discovered_cluster_manager`, for keeping compatibility when using `GET _cat/nodes?v&h=discovered_master` to show the specific column only.
- Add YAML rest test

Current response of 'GET Cat Health' API:
```
http://localhost:9200/_cat/health?v

epoch      timestamp cluster status node.total node.data discovered_master shards pri relo init unassign pending_tasks max_task_wait_time active_shards_percent
1646972456 04:20:56  runTask green           1         1              true      0   0    0    0        0             0                  -                100.0%
```

Proposed response of 'GET Cat Health' API:
```
http://localhost:9200/_cat/health?v

epoch      timestamp cluster status node.total node.data discovered_cluster_manager shards pri relo init unassign pending_tasks max_task_wait_time active_shards_percent
1646972456 04:20:56  runTask green           1         1                       true      0   0    0    0        0             0                  -                100.0%
```

**Explanation on the API response compatibility**
Because "cat API is a human-readable interface" (https://opensearch.org/docs/latest/opensearch/rest-api/cat/index/), and are not intended for use by software program, the old name "master" is not retained.

To keep the same API response with nodes in 1.x version, users can specify "header" in the request parameter, then the API response will remain the same for nodes in both 1.x and 2.x versions.
For example:

```
GET _cat/health?v&h=status&discovered_master

status discovered_master
green               true
```

**Testing**
```
./gradlew ':rest-api-spec:yamlRestTest' \
--tests "org.opensearch.test.rest.ClientYamlTestSuiteIT" \
-Dtests.method="test {p0=cat.health/10_basic/*}"
```

### Issues Resolved
Part of #1549 
 
### Check List
- [x] New functionality includes testing.
  - [ ] All tests pass
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
- [ ] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by tlfeng** [comment]
In log 3254:
```
> Task :server:internalClusterTest

REPRODUCE WITH: ./gradlew ':server:internalClusterTest' --tests "org.opensearch.blocks.SimpleBlocksIT.testAddBlockWhileDeletingIndices" -Dtests.seed=EF2BD3D9D42CE8CA -Dtests.security.manager=true -Dtests.jvm.argline="-XX:TieredStopAtLevel=1 -XX:ReservedCodeCacheSize=64m" -Dtests.locale=ar-LB -Dtests.timezone=Asia/Manila -Druntime.java=17

org.opensearch.blocks.SimpleBlocksIT > testAddBlockWhileDeletingIndices FAILED
    com.carrotsearch.randomizedtesting.UncaughtExceptionError: Captured an uncaught exception in thread: Thread[id=732, name=Thread-16, state=RUNNABLE, group=TGRP-SimpleBlocksIT]

        Caused by:
        java.lang.AssertionError: 
        Expected: an instance of org.opensearch.index.IndexNotFoundException
             but: <ProcessClusterEventTimeoutException[failed to process cluster event (delete-index [[glwvxoqqaa/i2W7ssjoTUeQ4iwAk6ARBQ]]) within 30s]> is a org.opensearch.cluster.metadata.ProcessClusterEventTimeoutException
            at __randomizedtesting.SeedInfo.seed([EF2BD3D9D42CE8CA]:0)
            at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:18)
            at org.junit.Assert.assertThat(Assert.java:964)
            at org.junit.Assert.assertThat(Assert.java:930)
            at org.opensearch.blocks.SimpleBlocksIT.lambda$testAddBlockWhileDeletingIndices$14(SimpleBlocksIT.java:483)
            at org.opensearch.blocks.SimpleBlocksIT.lambda$testAddBlockWhileDeletingIndices$15(SimpleBlocksIT.java:498)

    com.carrotsearch.randomizedtesting.UncaughtExceptionError: Captured an uncaught exception in thread: Thread[id=734, name=Thread-18, state=RUNNABLE, group=TGRP-SimpleBlocksIT]

        Caused by:
        java.lang.AssertionError: 
        Expected: an instance of org.opensearch.index.IndexNotFoundException
             but: <ProcessClusterEventTimeoutException[failed to process cluster event (delete-index [[pfoumeuzfk/SuaajRD-SWqAsgyGa_h31Q]]) within 30s]> is a org.opensearch.cluster.metadata.ProcessClusterEventTimeoutException
            at __randomizedtesting.SeedInfo.seed([EF2BD3D9D42CE8CA]:0)
            at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:18)
            at org.junit.Assert.assertThat(Assert.java:964)
            at org.junit.Assert.assertThat(Assert.java:930)
            at org.opensearch.blocks.SimpleBlocksIT.lambda$testAddBlockWhileDeletingIndices$14(SimpleBlocksIT.java:483)
            at org.opensearch.blocks.SimpleBlocksIT.lambda$testAddBlockWhileDeletingIndices$15(SimpleBlocksIT.java:498)
```

**Comment by dblock** [comment]
start gradle check

**Comment by tlfeng** [comment]
In log 3267:
```
> Task :qa:mixed-cluster:v1.3.0#mixedClusterTest

REPRODUCE WITH: ./gradlew ':qa:mixed-cluster:v1.3.0#mixedClusterTest' --tests "org.opensearch.backwards.MixedClusterClientYamlTestSuiteIT" -Dtests.method="test {p0=indices.get_field_mapping/20_missing_field/Return empty object if field doesn't exist, but index does}" -Dtests.seed=2E22A45097655A52 -Dtests.security.manager=true -Dtests.jvm.argline="-XX:TieredStopAtLevel=1 -XX:ReservedCodeCacheSize=64m" -Dtests.locale=pl -Dtests.timezone=AGT -Druntime.java=17

org.opensearch.backwards.MixedClusterClientYamlTestSuiteIT > test {p0=indices.get_field_mapping/20_missing_field/Return empty object if field doesn't exist, but index does} FAILED
    java.lang.AssertionError: Failure at [indices.get_field_mapping/20_missing_field:13]: expected [2xx] status code but api [indices.get_field_mapping] returned [404 Not Found] [{}]
        at __randomizedtesting.SeedInfo.seed([2E22A45097655A52:A6769B8A399937AA]:0)
        at org.opensearch.test.rest.yaml.OpenSearchClientYamlSuiteTestCase.executeSection(OpenSearchClientYamlSuiteTestCase.java:442)
        at org.opensearch.test.rest.yaml.OpenSearchClientYamlSuiteTestCase.test(OpenSearchClientYamlSuiteTestCase.java:415)
...
```

**Comment by tlfeng** [comment]
In log 3306:
```
> Task :qa:mixed-cluster:v1.3.0#mixedClusterTest

REPRODUCE WITH: ./gradlew ':qa:mixed-cluster:v1.3.0#mixedClusterTest' --tests "org.opensearch.backwards.MixedClusterClientYamlTestSuiteIT" -Dtests.method="test {p0=cat.health/10_basic/Help}" -Dtests.seed=C90F9A5215A51187 -Dtests.security.manager=true -Dtests.jvm.argline="-XX:TieredStopAtLevel=1 -XX:ReservedCodeCacheSize=64m" -Dtests.locale=es-UY -Dtests.timezone=Asia/Katmandu -Druntime.java=17

org.opensearch.backwards.MixedClusterClientYamlTestSuiteIT > test {p0=cat.health/10_basic/Help} FAILED
    java.lang.AssertionError: Failure at [cat.health/10_basic:10]: field [$body] was expected to match the provided regex but didn't
    Expected: ^  epoch         .+ \n
        timestamp     .+ \n
        cluster       .+ \n
        status        .+ \n
        node.total    .+ \n
        node.data     .+ \n
        discovered_master .+ \n
        shards        .+ \n
        pri           .+ \n
        relo          .+ \n
        init          .+ \n
        unassign      .+ \n
        pending_tasks .+ \n
        max_task_wait_time .+ \n
        active_shards_percent .+ \n

    $
         but: was "epoch                      | t,time                                   | seconds since 1970-01-01 00:00:00   \ntimestamp                  | ts,hms,hhmmss                            | time in HH:MM:SS                    \ncluster                    | cl                                       | cluster name                        \nstatus                     | st                                       | health status                       \nnode.total                 | nt,nodeTotal                             | total number of nodes               \nnode.data                  | nd,nodeData                              | number of nodes that can store data \ndiscovered_cluster_manager | dcm,dm,discovered_master                 | cluster manager is discovered or not\nshards                     | t,sh,shards.total,shardsTotal            | total number of shards              \npri                        | p,shards.primary,shardsPrimary 
...
```

**Comment by tlfeng** [comment]
In log 3312:
```
> Task :qa:mixed-cluster:v1.4.0#mixedClusterTest

REPRODUCE WITH: ./gradlew ':qa:mixed-cluster:v1.4.0#mixedClusterTest' --tests "org.opensearch.backwards.MixedClusterClientYamlTestSuiteIT" -Dtests.method="test {p0=cat.health/10_basic/Help}" -Dtests.seed=81BC6F0247CA4A37 -Dtests.security.manager=true -Dtests.jvm.argline="-XX:TieredStopAtLevel=1 -XX:ReservedCodeCacheSize=64m" -Dtests.locale=ru-RU -Dtests.timezone=Pacific/Kwajalein -Druntime.java=17

org.opensearch.backwards.MixedClusterClientYamlTestSuiteIT > test {p0=cat.health/10_basic/Help} FAILED
    java.lang.AssertionError: Failure at [cat.health/10_basic:10]: field [$body] was expected to match the provided regex but didn't
    Expected: ^  epoch         .+ \n
        timestamp     .+ \n
        cluster       .+ \n
        status        .+ \n
        node.total    .+ \n
        node.data     .+ \n
        discovered_master .+ \n
        shards        .+ \n
        pri           .+ \n
        relo          .+ \n
        init          .+ \n
        unassign      .+ \n
        pending_tasks .+ \n
        max_task_wait_time .+ \n
        active_shards_percent .+ \n

    $
         but: was "epoch                      | t,time                                   | seconds since 1970-01-01 00:00:00   \ntimestamp                  | ts,hms,hhmmss                            | time in HH:MM:SS                    \ncluster                    | cl                                       | cluster name                        \nstatus                     | st                                       | health status                       \nnode.total                 | nt,nodeTotal                             | total number of nodes               \nnode.data                  | nd,nodeData                              | number of nodes that can store data \ndiscovered_cluster_manager | dcm,dm,discovered_master                 | cluster manager is discovered or not\nshards                     | t,sh,shards.total,shardsTotal            | total number of shards              \npri                        | p,shards.primary,shardsPrimary           | number of primary shards            \nrelo                       | r,shards.relocating,shardsRelocating     | number of relocating nodes          \ninit                       | i,shards.initializing,shardsInitializing | number of initializing nodes        \nunassign                   | u,shards.unassigned,shardsUnassigned     | number of unassigned shards         \npending_tasks              | pt,pendingTasks                          | number of pending tasks             \nmax_task_wait_time         | mtwt,maxTaskWaitTime                     | wait time of longest task pending   \nactive_shards_percent      | asp,activeShardsPercent                  | active number of shards in percent  \n"
```
```
REPRODUCE WITH: ./gradlew ':server:internalClusterTest' --tests "org.opensearch.gateway.RecoveryFromGatewayIT.testReuseInFileBasedPeerRecovery" -Dtests.seed=81BC6F0247CA4A37 -Dtests.security.manager=true -Dtests.jvm.argline="-XX:TieredStopAtLevel=1 -XX:ReservedCodeCacheSize=64m" -Dtests.locale=is -Dtests.timezone=Europe/Skopje -Druntime.java=17

org.opensearch.gateway.RecoveryFromGatewayIT > testReuseInFileBasedPeerRecovery FAILED
    java.lang.AssertionError: shard [test][0] on node [node_t1] has pending operations:
     --> RetentionLeaseBackgroundSyncAction.Request{retentionLeases=RetentionLeases{primaryTerm=1, version=1547, leases={peer_recovery/oyPwOGGkQNW8Y4rF2xKeWQ=RetentionLease{id='peer_recovery/oyPwOGGkQNW8Y4rF2xKeWQ', retainingSequenceNumber=713, timestamp=1647158126249, source='peer recovery'}, peer_recovery/twOZnWW1QNqDa4-ktwGYig=RetentionLease{id='peer_recovery/twOZnWW1QNqDa4-ktwGYig', retainingSequenceNumber=713, timestamp=1647158126249, source='peer recovery'}}}, shardId=[test][0], timeout=1m, index='test', waitForActiveShards=0}
    	at org.opensearch.index.shard.IndexShardOperationPermits.acquire(IndexShardOperationPermits.java:248)
    	at org.opensearch.index.shard.IndexShard.acquirePrimaryOperationPermit(IndexShard.java:3130)
```
It's reported in issue https://github.com/opensearch-project/OpenSearch/issues/1746

**Comment by tlfeng** [comment]
In log 3325:
```
> Task :rest-api-spec:yamlRestTest

REPRODUCE WITH: ./gradlew ':rest-api-spec:yamlRestTest' --tests "org.opensearch.test.rest.ClientYamlTestSuiteIT" -Dtests.method="test {p0=cat.health/10_basic/Help}" -Dtests.seed=509C299162A8ADE4 -Dtests.security.manager=true -Dtests.jvm.argline="-XX:TieredStopAtLevel=1 -XX:ReservedCodeCacheSize=64m" -Dtests.locale=he -Dtests.timezone=America/St_Johns -Druntime.java=17

org.opensearch.test.rest.ClientYamlTestSuiteIT > test {p0=cat.health/10_basic/Help} FAILED
    java.lang.AssertionError: Failure at [cat.health/10_basic:47]: field [$body] was expected to match the provided regex but didn't
    Expected: ^  epoch         .+ \n
        timestamp     .+ \n
        cluster       .+ \n
        status        .+ \n
        node.total    .+ \n
        node.data     .+ \n
        discovered_master .+ \n
        shards        .+ \n
        pri           .+ \n
        relo          .+ \n
        init          .+ \n
        unassign      .+ \n
        pending_tasks .+ \n
        max_task_wait_time .+ \n
        active_shards_percent .+ \n

    $
         but: was "epoch                      | t,time                                   | seconds since 1970-01-01 00:00:00   \ntimestamp                  | ts,hms,hhmmss                            | time in HH:MM:SS                    \ncluster                    | cl                                       | cluster name                        \nstatus                     | st                                       | health status                       \nnode.total                 | nt,nodeTotal                             | total number of nodes               \nnode.data                  | nd,nodeData                              | number of nodes that can store data \ndiscovered_cluster_manager | dcm,dm,discovered_master         
...
```
```
> Task :qa:smoke-test-multinode:integTest

REPRODUCE WITH: ./gradlew ':qa:smoke-test-multinode:integTest' --tests "org.opensearch.smoketest.SmokeTestMultiNodeClientYamlTestSuiteIT" -Dtests.method="test {yaml=cat.health/10_basic/Help}" -Dtests.seed=509C299162A8ADE4 -Dtests.security.manager=true -Dtests.jvm.argline="-XX:TieredStopAtLevel=1 -XX:ReservedCodeCacheSize=64m" -Dtests.locale=en-PH -Dtests.timezone=Etc/GMT-0 -Druntime.java=17

org.opensearch.smoketest.SmokeTestMultiNodeClientYamlTestSuiteIT > test {yaml=cat.health/10_basic/Help} FAILED
    java.lang.AssertionError: Failure at [cat.health/10_basic:47]: field [$body] was expected to match the provided regex but didn't
    Expected: ^  epoch         .+ \n
        timestamp     .+ \n
        cluster       .+ \n
        status        .+ \n
        node.total    .+ \n
        node.data     .+ \n
        discovered_master .+ \n
```

**Comment by tlfeng** [comment]
In log 3333:
Seems not related to the code changes.
```
> Task :server:internalClusterTest

REPRODUCE WITH: ./gradlew ':server:internalClusterTest' --tests "org.opensearch.indices.state.CloseWhileRelocatingShardsIT.testCloseWhileRelocatingShards" -Dtests.seed=25C5B40A42AAB4CC -Dtests.security.manager=true -Dtests.jvm.argline="-XX:TieredStopAtLevel=1 -XX:ReservedCodeCacheSize=64m" -Dtests.locale=hu-HU -Dtests.timezone=Africa/Addis_Ababa -Druntime.java=17

org.opensearch.indices.state.CloseWhileRelocatingShardsIT > testCloseWhileRelocatingShards FAILED
    java.lang.Exception: Test abandoned because suite timeout was reached.
        at __randomizedtesting.SeedInfo.seed([25C5B40A42AAB4CC]:0)

... (lots of test failures)

        Caused by:
        java.lang.OutOfMemoryError: Java heap space
```
```
Tests with failures:
 - org.opensearch.indices.state.CloseWhileRelocatingShardsIT.testCloseWhileRelocatingShards
 - org.opensearch.indices.state.CloseWhileRelocatingShardsIT.classMethod

3637 tests completed, 1038 failed, 48 skipped
```

**Comment by tlfeng** [comment]
start gradle check

File: server/src/main/java/org/opensearch/rest/action/cat/RestHealthAction.java [hunk 1]
```diff
@@ -88,7 +88,12 @@ protected Table getTableWithHeader(final RestRequest request) {
         t.addCell("status", "alias:st;desc:health status");
         t.addCell("node.total", "alias:nt,nodeTotal;text-align:right;desc:total number of nodes");
         t.addCell("node.data", "alias:nd,nodeData;text-align:right;desc:number of nodes that can store data");
-        t.addCell("discovered_master", "alias:dm;text-align:right;desc:discovered master");
```
Comment (kartg) [inline server/src/main/java/org/opensearch/rest/action/cat/RestHealthAction.java:91]: With the Cluster Health API - #2437 - we chose to _add_ a "cluster_manager" field alongside the "master" one, but here we're _replacing_ "master" with "cluster_manager". Why the difference in approach?

Comment (kartg) [reply server/src/main/java/org/opensearch/rest/action/cat/RestHealthAction.java:91]: D'oh, should've read the PR description more carefully 😅 

To answer my own question - #2437 is adding a new field in the response, so we cannot replace it in-line. In this PR, the change affects the human-readable table header in the output, which we hardly expect to be parsed by code.

Feel free to ignore my question above 😄

**Review by kartg — APPROVED** [review]

File: server/src/main/java/org/opensearch/rest/action/cat/RestHealthAction.java (hunk 1 above)
Comment (tlfeng) [reply server/src/main/java/org/opensearch/rest/action/cat/RestHealthAction.java:91]: Hi @kartg, I really appreciate your careful attitude, to put the explanation in detail.
It's also my negligence that I didn't explain more in this PR description, after add more description in anther PR with the same concern: https://github.com/opensearch-project/OpenSearch/pull/2441#issuecomment-1068547454 .
I will update in this PR description to encourage users who wants to use the same API request to parse the response of CAT Health in 2.x nodes, to filter output by specifying table header `GET _cat/health?v&h=discovered_master`

**Review by Poojita-Raj — COMMENTED** [review]
LGTM!

**Pull request merged**